lib/discourse_translator/guardian_extension.rb

```diff
@@ -1,25 +1,12 @@
 # frozen_string_literal: true
 module DiscourseTranslator::GuardianExtension
-  def user_group_allowed?
-    authorized_groups = SiteSetting.restrict_translation_by_group.split("|").map(&:to_i)
-
-    authorized? current_user, authorized_groups
+  def user_group_allow_translate?
+    !!current_user&.in_any_groups?(SiteSetting.restrict_translation_by_group_map)
   end
 
-  def post_group_allowed?(post)
-    authorized_poster_groups =
-      SiteSetting.restrict_translation_by_poster_group.split("|").map(&:to_i)
-    return true if authorized_poster_groups.empty?
-
-    poster = User.find post.user_id
-    authorized? poster, authorized_poster_groups
-  end
-
-  def authorized?(user, authorized_groups)
-    return false if !user
-
-    user_groups = user.groups.pluck :id
-    authorized = authorized_groups.intersection user_groups
-    !authorized.empty?
+  def poster_group_allow_translate?(post)
+    return false if !current_user
+    return true if SiteSetting.restrict_translation_by_poster_group_map.empty?
+    !!post.user.in_any_groups?(SiteSetting.restrict_translation_by_poster_group_map)
```

Comment:
Hmm maybe I am just missing something with the `!!` usage, can you please explain it? It's kind of confusing to read

Reply:
Weird. I'd noticed that the function was returning a `nil`. Will check.

Reply:
Ah, it was the usage of `current_user&.` which was causing it. The tests were testing for anon user so I decided to keep it so we don't make assumptions about usage.